automated/linux/xfstests/xfstests.sh

```diff
@@ -0,0 +1,175 @@
+
+OUTPUT="$(pwd)/output"
+RESULT_FILE="${OUTPUT}/result.txt"
+export RESULT_FILE
+
+# Results logs
+RESULT_LOG="${OUTPUT}/logs.txt"
+RESULT_PASS="${OUTPUT}/pass.txt"
+RESULT_FAIL="${OUTPUT}/fail.txt"
+RESULT_SKIP="${OUTPUT}/skip.txt"
+
+SKIP_INSTALL="false"
+
+# xfstests test build path
+XFSTESTS_PATH="/opt/xfstests"
+
+TEST_IMG=test.img
+SCRATCH_IMG=scratch.img
+TEST_DEV=/dev/loop0
+SCRATCH_DEV=/dev/loop1
+TEST_DIR=/mnt/test
+SCRATCH_MNT=/mnt/scratch
+FILESYSTEM="ext4"
+T_SIZE="5G"
+S_SIZE="8G"
+
+# Print usage
+usage() {
+    info_msg "Usage: $0 [-d </dev/sdb>] [-e </dev/loop0>]
+                           [-f <ext4>] [-m </mnt/scratch>]
+                           [-t </mnt/test>] [-s <true|false>]
+                           [-x <10G>] [-z <10G>] [-h]
+
+    -d <device>     Specify the test device path (default: /dev/loop0)
+    -e <device>     Specify the scratch device path (default: /dev/loop1)
+    -f <filesystem> Set the filesystem type (default: ext4)
+    -m <path>       Set the scratch mount path (default: /mnt/scratch)
+    -t <path>       Set the test mount path (default: /mnt/test)
+    -s <true>       Skip package installation (default: false)
+    -x <size>       Set the test and scratch size (default: 5G for test, 8G for scratch)
+    -z <size>       Set the test and scratch size (default: 5G for test, 8G for scratch)
+    -h              Show this help message and exit
+" 1>&2
+    exit 1
+}
+
+results_parser() {
+    # Parse pass test cases
+    find results/*/*.full -print0 | awk -v RS='\0' -F'/' '{print $2"-"$3" pass"}' | sed 's/.full$//' >> "${RESULT_PASS}"
+    # Parse fail test cases
+    find results/*/*.out.bad -print0 | awk -v RS='\0' -F'/' '{print $2"-"$3" fail"}' | sed 's/.out.bad$//' >> "${RESULT_FAIL}"
+    # Parse skip test cases
+    find results/*/*.notrun -print0 | awk -v RS='\0' -F'/' '{print $2"-"$3" skip"}' | sed 's/.notrun$//' >> "${RESULT_SKIP}"
+    # Append all the results to results.txt file
+    cat "${RESULT_PASS}" "${RESULT_FAIL}" "${RESULT_SKIP}" 2>&1 | tee -a "${RESULT_FILE}"
+}
+
+# test setup
+test_setup() {
+    export TEST_IMG="${TEST_IMG}"
+    export SCRATCH_IMG="${SCRATCH_IMG}"
+    export TEST_DEV="${TEST_DEV}"
+    export SCRATCH_DEV="${SCRATCH_DEV}"
+    export TEST_DIR="${TEST_DIR}"
+    export SCRATCH_MNT="${SCRATCH_MNT}"
+    export FILESYSTEM="${FILESYSTEM}"
+}
+
+# run_xfstests ext4
+run_xfstests() {
+    filesystem="$1"
+    info_msg "run xfstests : ${filesystem}"
+    # export required configs
+    test_setup
+    # print disk space usage
+    df
+    # print mount
+    mount
+    if [ "${filesystem}" = "xfs" ]; then
+        ./check -g "${filesystem}"/quick -x dmapi 2>&1 | tee -a "${RESULT_LOG}"
+    elif [ "${filesystem}" = "ext2" ]; then
+        ./check -g generic 2>&1 | tee -a "${RESULT_LOG}"
+    elif [ "${filesystem}" = "ext3" ]; then
+        ./check -g generic 2>&1 | tee -a "${RESULT_LOG}"
+    else
+        ./check -g "${filesystem}"/quick 2>&1 | tee -a "${RESULT_LOG}"
```

Comment:
Don't we want ext4 generic here?

Reply:
@bhcopeland Good question.
We will do incremental test coverage development. 
ext4 generic will be addressed in the next phase.
Because I have seen a collect of generic test failures. we need to investigate and enable them in incremental way.